query-engine/connectors/sql-query-connector/src/database/js.rs

```diff
@@ -13,6 +13,8 @@ use quaint::{
 };
 use std::sync::{Arc, Mutex};
 
+// TODO: evaluate turning this into `Lazy<Mutex<Option<Arc<DriverAdapter>>>>` to avoid
+// a clone+drop on the adapter passed via `Js::from_source`.
```

Comment:
TODO: mention that this, for the time being, would cause symbol linking errors due to Napi.rs, as Miguel mentioned down below.